Only swap when holding shift for banking and shopping

Infinitay opened the pull request:
- Only swap withdraw/deposit when holding shift
- Only swap buy/sell when holding shift
- Add method to clear all swaps `MenuManager#clearSwaps`
- Removed `MenuEntrySwapperPlugin#removeSwaps`, replaced with `MenuManager#clearSwaps`
- Update buy/sell swaps to swap with value `MenuEntry `to allow for one clicking
- Updated config for new changes, made banking/shopping config a little more clear/convenient 

I should note I haven't tested every single change, but I have tested burning amulet after the remove swap change, as well as shift-only banking/shopping and disabled shift-only banking/shopping.

Ganom commented:
If you clear all swaps, it will remove other plugin swaps aswell, which might be unsafe.

Infinitay commented:
> If you clear all swaps, it will remove other plugin swaps aswell, which might be unsafe. 

Good point, might want to refrain from pushing to master then. However, I'd assume most other plugins that depend on adding swaps do so _dynamically_ for a lack of better terms, as in, they're added on game ticks or some other event. For example the blackjack plugin, it adds to the swap map using the `MenuEntryAdded` event.

So I would assume it wouldn't be a problem?

Infinitay commented:
As of new injector update, this isn't working. I will figure out whats wrong and update the PR

Infinitay commented:
Fixed all known bugs that existed in this PR (https://github.com/runelite-extended/runelite/pull/491#issuecomment-501810982)

Additional changes to note:
- Renamed config from withdraw to bank
  - Mitigate old settings to new config after config rename
- Renamed MenuManager#clearSwaps to clearAllSwaps
  - Added clearing priority swap entries to clearAllSwaps

Infinitay commented:
Forgot to push new changes, I added regex support for banking/shopping swaps.

One thing that you should know is that `*` is handled as `.*` to try and make it a little user friendly for people not too familiar with regex. The all item patterns are pre-compiled on start-up, or just a certain item list when edited (ie change bank-all items, then only bank-all items would be compiled) for performance. If there is a better way to do it, do let me know, but I think this is the best way.

The one thing I would like to do is add the ability for priorities. For example, you could have wildcard for bank-all items (`*`), but under bank-10 you could have `Shark`. With the following, the bank-10 entry would have priority over bank-all. Think of it as if the `target` matches multiple regex Sets, then it'll swap according to the second _highest_ set. Although, I don't know how to go about it at this time, and it's not really my priority.
